py3status/modules/uptime.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+"""
+Display the system's uptime
+
+Configuration parameters:
+        - format : what will be displayed
+
+@author Alexis "Horgix" Chotard <alexis.horgix.chotard@gmail.com>
+@license BSD
+"""
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    format = "{days} days, {hours} hours, {minutes} min"
+
+    def _uptime(self):
+        """Returns uptime in seconds or None, on Linux."""
+        with open('/proc/uptime', 'r') as f:
+            up = float(f.readline().split()[0])
+        return up
+
+    def uptime(self, i3s_output_list, i3s_config):
+        response = {}
+
+        up = self._uptime()
+        # Computing from bare seconds since timedelta only provides .days and
+        # .seconds anyway
+        days, up = divmod(up, 86400)
+        hours, up = divmod(up, 3600)
+        minutes, up = divmod(up, 60)
+        seconds = up
```

Comment:
I don't think that's necessary.
On Mar 7, 2016 10:56, "Horgix" <notifications@github.com> wrote:

> In py3status/modules/uptime.py
> <https://github.com/ultrabug/py3status/pull/205#discussion_r55233663>:
>
> > +    def _uptime(self):
> > +        """Returns uptime in seconds or None, on Linux."""
> > +        with open('/proc/uptime', 'r') as f:
> > +            up = float(f.readline().split()[0])
> > +        return up
> > +
> > +    def uptime(self, i3s_output_list, i3s_config):
> > +        response = {}
> > +
> > +        up = self._uptime()
> > +        # Computing from bare seconds since timedelta only provides .days and
> > +        # .seconds anyway
> > +        days, up = divmod(up, 86400)
> > +        hours, up = divmod(up, 3600)
> > +        minutes, up = divmod(up, 60)
> > +        seconds = up
>
> @pydsigner <https://github.com/pydsigner>
>
> Looks like the docstring for _uptime() is just misleading.
>
> Removed _uptime anyway. Should I add a docstring to the main uptime()
> function considering that it already is documented in the "main docstring"
> for the module ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ultrabug/py3status/pull/205/files#r55233663>.
>